wfe/web-front-end.go

```diff
@@ -968,29 +989,42 @@ func (wfe *WebFrontEndImpl) Authorization(response http.ResponseWriter, request
 
 	// If there is a fragment, then this is actually a request to a challenge URI
 	if len(request.URL.RawQuery) != 0 {
-		logEvent = wfe.challenge(authz, response, request, logEvent)
-		return
+		wfe.challenge(response, request, authz, &logEvent)
```

Comment:
Reminder to self: we should clean this up.

We can drop pat in here to do that. I've got a v old branch to do some of that I can repurpose, maybe.